lib/matrix/matrix.nit

```diff
@@ -293,3 +307,30 @@ class MatrixCoordinate
 
 	redef fun to_s do return "({x},{y})"
 end
+
+# Dedicated native structure to avoid boxing cost
+private extern class NativeMatrix `{ double* `}
+
+	new(buffer: CString) `{ return (double*)buffer; `}
```

Comment:
I'm not sure why `new` takes a CString. I expected more a `capacity: Int` as parameter.
Moreover, why using a CString?